cfpb/sbl-frontend
# [Filing] Route protection

Close #582 

## Changes

- Fix: FilingCreate - Improve method of redirection
- Feat: FilingSteps - Filing route protection
  - If a user tries to access a Filing step beyond where they've progressed in the Filing process, we redirect them to the next step they need to focus on.


## How to test this PR

1. Run through the Filing process for an Institution that has not upload before and verify you can get to the `Submit` stage
1. Run through the Filing process for an Institution that has previously uploaded before and verify you can get to the `Submit` stage
2. Try to go directly reach later stages for an institution that has not reached those stages (i.e. `/submit` for an institution who needs to correct `/errors`) and verify that you are redirected to the appropriate Filing step

## Demo
- User is on the `warnings` step
- User tries to reach the `submit` step but is redirect to `warnings`
- User tires to reach the `contact` step but is redirect to `warnings`
- User is able to directly access the `errors` step since it is complete
- User is able to directly access the `upload` step since it is complete

https://github.com/cfpb/sbl-frontend/assets/2592907/ce4c6065-78ac-4f32-a000-8e3a31597cd1
